src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -75,49 +75,71 @@ function DiscoverDocumentsComponent({
   onAddFilter?: DocViewFilterFn;
   savedSearch: SavedSearch;
   setExpandedDoc: (doc?: DataTableRecord) => void;
-  state: AppState;
-  stateContainer: GetStateReturn;
+  stateContainer: DiscoverStateContainer;
   onFieldEdited?: () => void;
 }) {
   const { capabilities, dataViews, uiSettings } = useDiscoverServices();
+  const [query, sort, rowHeight, rowsPerPage, grid, columns, index] = useAppStateSelector(
+    (state) => {
+      return [
+        state.query,
+        state.sort,
+        state.rowHeight,
+        state.rowsPerPage,
+        state.grid,
+        state.columns,
+        state.index,
+      ];
+    }
+  );
+
   const useNewFieldsApi = useMemo(() => !uiSettings.get(SEARCH_FIELDS_FROM_SOURCE), [uiSettings]);
   const hideAnnouncements = useMemo(() => uiSettings.get(HIDE_ANNOUNCEMENTS), [uiSettings]);
   const isLegacy = useMemo(() => uiSettings.get(DOC_TABLE_LEGACY), [uiSettings]);
   const sampleSize = useMemo(() => uiSettings.get(SAMPLE_SIZE_SETTING), [uiSettings]);
 
   const documentState = useDataState(documents$);
   const isLoading = documentState.fetchStatus === FetchStatus.LOADING;
-  const isPlainRecord = useMemo(
-    () => getRawRecordType(state.query) === RecordRawType.PLAIN,
-    [state.query]
-  );
+  // this is needed to prevent data view pushing onSort because there has been a state change with a new data view
```

Comment:
I'd like to walk through the code so you can show me this. It sounds like a good test of whether I understand this code. Also, it sounds like a flaw in the state model.

Reply:
I've improved documentation why this is written that way .. generally it's fixing a situation when switching between data views with different timestamps, by showing the loading state for this cases. Not saying this is ideal. I've improved the documentation for this, and happy to have a chat about it for sure 👍